packages/itwinui-react/src/core/DropdownMenu/DropdownMenu.tsx

```diff
@@ -87,6 +98,14 @@ const DropdownMenuContent = React.forwardRef((props, forwardedRef) => {
     onVisibleChange,
   );
 
+  const middleware = React.useMemo(
+    () => ({
+      hide: false,
```

Comment:
I was actually hoping that we would enable `hide` by default. #2235 is clearly a bug, and we shouldn't require users to opt-in; it should just do the right thing by default. And since `hide` has been problematic in the past, we can still allow users to opt-out when it actually causes issues.